graph/pull_v2.go

```diff
@@ -171,24 +179,45 @@ func (p *v2Puller) download(di *downloadInfo) {
 		ID:        stringid.TruncateID(di.img.ID),
 		Action:    "Downloading",
 	})
-	io.Copy(tmpFile, reader)
+
+	downloadComplete := make(chan error)
+	readCanceler := ioutils.NewReadCloserWithError(reader)
```

Comment:
@jlhawn why not use ioutil.newreadcloserwitherror in lieu of ioutil.NopCloser here https://github.com/docker/docker/pull/15322/files#diff-7e2a216082b6bd557c7ecebd68305e2eR174 ?